pkg/ocm/share/repository/json/json.go

```diff
@@ -0,0 +1,508 @@
+		var s ocm.ReceivedShare
+		if err := utils.UnmarshalJSONToProtoV1(v, &s); err != nil {
+			return err
+		}
+		received[k] = &s
+	}
+
+	*s = shareModel{
+		Shares:         share,
+		ReceivedShares: received,
+	}
+
+	return nil
+}
+
+func (s *shareModel) MarshalJSON() ([]byte, error) {
+	shares := map[string]json.RawMessage{}
+	for k, v := range s.Shares {
+		d, err := utils.MarshalProtoV1ToJSON(v)
+		if err != nil {
+			return nil, err
+		}
+		shares[k] = d
+	}
+
+	received := map[string]json.RawMessage{}
+	for k, v := range s.ReceivedShares {
+		d, err := utils.MarshalProtoV1ToJSON(v)
+		if err != nil {
+			return nil, err
+		}
+		received[k] = d
+	}
+
+	return json.Marshal(map[string]any{
+		"shares":          shares,
+		"received_shares": received,
+	})
+}
+
+type config struct {
+	File string `mapstructure:"file"`
+}
+
+func (c *config) ApplyDefaults() {
+	if c.File == "" {
+		c.File = "/var/tmp/reva/ocm-shares.json"
+	}
+}
+
+type mgr struct {
+	c          *config
+	sync.Mutex // concurrent access to the file
+	model      *shareModel
+}
+
+func (m *mgr) save() error {
+	f, err := os.OpenFile(m.c.File, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return errors.Wrap(err, "error opening file "+m.c.File)
+	}
+	defer f.Close()
+
+	if err := json.NewEncoder(f).Encode(m.model); err != nil {
+		return errors.Wrap(err, "error encoding to json")
+	}
+
+	return f.Sync()
+}
+
+func (m *mgr) load() error {
+	f, err := os.OpenFile(m.c.File, os.O_RDONLY, 0644)
+	if err != nil {
+		return errors.Wrap(err, "error opening file "+m.c.File)
+	}
+	defer f.Close()
+
+	d, err := io.ReadAll(f)
+	if err != nil {
+		return err
+	}
+
+	var model shareModel
+	if err := json.Unmarshal(d, &model); err != nil {
+		return err
+	}
+
+	m.model = &model
+	return nil
+}
+
+func genID() string {
+	return uuid.New().String()
+}
+
+func (m *mgr) StoreShare(ctx context.Context, ocmshare *ocm.Share) (*ocm.Share, error) {
+	m.Lock()
+	defer m.Unlock()
+
+	if err := m.load(); err != nil {
+		return nil, err
+	}
+
+	if _, err := m.getByKey(ctx, &ocm.ShareKey{
+		Owner:      ocmshare.Owner,
+		ResourceId: ocmshare.ResourceId,
+		Grantee:    ocmshare.Grantee,
+	}); err == nil {
+		return nil, share.ErrShareAlreadyExisting
+	}
+
+	ocmshare.Id = &ocm.ShareId{OpaqueId: genID()}
+	m.model.Shares[ocmshare.Id.OpaqueId] = cloneShare(ocmshare)
+
+	if err := m.save(); err != nil {
+		return nil, errors.Wrap(err, "error saving share")
+	}
+
+	return ocmshare, nil
+}
+
+func cloneShare(s *ocm.Share) *ocm.Share {
+	d, err := utils.MarshalProtoV1ToJSON(s)
+	if err != nil {
+		panic(err)
+	}
+	var cloned ocm.Share
+	if err := utils.UnmarshalJSONToProtoV1(d, &cloned); err != nil {
+		panic(err)
+	}
+	return &cloned
+}
+
+func cloneReceivedShare(s *ocm.ReceivedShare) *ocm.ReceivedShare {
+	d, err := utils.MarshalProtoV1ToJSON(s)
+	if err != nil {
+		panic(err)
+	}
+	var cloned ocm.ReceivedShare
+	if err := utils.UnmarshalJSONToProtoV1(d, &cloned); err != nil {
+		panic(err)
```

Comment:
Can we ignore the error instead of panicing?

Reply:
Those errors are gonna be handled now.